code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -525,6 +525,15 @@
 		desc = "Your standard drinking glass."
 		return
 
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/pen))
+		var/txt = stripped_input(user, "What would you like to name this drink?", "Mixology", "", 30)
+		if(txt)
```

Comment:
Ah Yes. Keeping the `input` open and renaming glasses as a ghost. That sure is great.